recipes-browser/cog/cog.inc

```diff
@@ -40,10 +40,14 @@ FILES_SOLIBSDEV = "${libdir}/libcogcore*.so"
 FILES_${PN} += "${libdir}/libcogplatform*.so"
 INSANE_SKIP_${PN} = "dev-so"
 
+# Local user name starting cog with DBus system session enabled. This should be
+# set only if the dbus PACKAGECONFIG is enabled.
+COG_DBUS_OWN_USER ?= ""
+
 # Use WebKitGTK+ instead of WPEWebKit
 PACKAGECONFIG[webkitgtk] = "-DCOG_USE_WEBKITGTK=ON,-DCOG_USE_WEBKITGTK=OFF"
 # Expose remote control interface on system bus.
-PACKAGECONFIG[dbus] = "-DCOG_DBUS_SYSTEM_BUS=ON,-DCOG_DBUS_SYSTEM_BUS=OFF"
+PACKAGECONFIG[dbus] = "-DCOG_DBUS_SYSTEM_BUS=ON -DCOG_DBUS_OWN_USER=${COG_DBUS_OWN_USER},-DCOG_DBUS_SYSTEM_BUS=OFF"
```

Comment:
What happens if the `COG_DBUS_OWN_USER=""`. If there is going to be configuration error?

Reply:
No. Cog will just fail to start at runtime.

Reply:
Not a expert. does it make sense to set this together with COG_DBUS_SYSTEM_BUS=ON?

Reply:
This is exactly what this patch does :)